packet.go

```diff
@@ -86,7 +87,16 @@ func readTagAndLength(conn *bufio.Reader, bytes *[]byte) (ret ldap.TagAndLength,
 	//	}
 	// We are expecting the LDAP sequence tag 0x30 as first byte
 	if b != 0x30 {
-		panic(fmt.Sprintf("Expecting 0x30 as first byte, but got %#x instead", b))
+		// Did not get 0x30 as first byte, which is invalid for an LDAP connection.
+		// Read the full packet and return an error instead
+		allBytes, err := io.ReadAll(conn)
```

Comment:
ReadAll is going to block until it sees EOF i.e. the client closes its side of the connection. And the client isn't going to do that because it's expecting a response from us. So I think you just want to make a single Read() call here. This is gonna give you all the data from the first packet, I think.

```
payload := make([]byte, 10000)
n, err := conn.Read(payload)
payload = payload[:n]
```